packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,273 @@
+
+  /** Text placeholder for the input field. */
+  placeholder?: string;
+
+  /** A render function to customize how cells are rendered in the Calendar. */
+  renderCell?: ShorthandRenderFunction<any>;
+
+  /** A render function to customize how cells are rendered in the Calendar.. */
+  renderHeaderCell?: ShorthandRenderFunction<any>;
+
+  /** Localized labels */
+  localizedStrings?: IDateGridStrings;
+}
+
+export type DatepickerStylesProps = never;
+
+export const datepickerClassName = 'ui-datepicker';
+
+const Unstable_Datepicker: ComponentWithAs<'div', DatepickerProps> &
+  FluentComponentStaticProps<DatepickerProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(Unstable_Datepicker.displayName, context.telemetry);
+  setStart();
+  const datepickerRef = React.useRef<HTMLElement>();
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [selectedDate, setSelectedDate] = React.useState<Date | undefined>();
+  const valueFormatter = date => (date ? formatMonthDayYear(date, DEFAULT_STRINGS) : '');
+  const { firstDayOfWeek, firstWeekOfYear, dateRangeType } = props;
+  const calendarOptions: IDayGridOptions = {
+    selectedDate: selectedDate ?? props.today ?? new Date(),
+    navigatedDate: selectedDate ?? props.today ?? new Date(),
+    firstDayOfWeek,
+    firstWeekOfYear,
+    dateRangeType,
+  };
+
+  const showCalendarGrid = () => {
+    setOpen(true);
+  };
+
+  const { className, design, styles, variables } = props;
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(Unstable_Datepicker.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: Unstable_Datepicker.displayName,
+    actionHandlers: {},
+    rtl: context.rtl,
+  });
+
+  const { classes } = useStyles<DatepickerStylesProps>(Unstable_Datepicker.displayName, {
+    className: datepickerClassName,
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
+
+  const handleChange: DatepickerCalendarProps['onDaySelect'] = (e, data) => {
+    const targetDay = data.value;
+    setSelectedDate(targetDay.originalDate);
+    setOpen(false);
+
+    _.invoke(props, 'onDayChange', e, { ...props, value: targetDay });
+  };
+
+  const element = (
+    <Ref innerRef={datepickerRef}>
+      {getA11yProps.unstable_wrapWithFocusZone(
+        <ElementType
+          {...getA11yProps('root', {
+            className: classes.root,
+            ...unhandledProps,
+          })}
+        >
+          <Input readOnly onClick={showCalendarGrid} value={valueFormatter(selectedDate)} />
+          <Popup
+            open={open}
+            onOpenChange={(e, { open }) => setOpen(open)}
+            content={
+              <DatepickerCalendar {...calendarOptions} onDaySelect={handleChange} localizedStrings={DEFAULT_STRINGS} />
```

Comment:
I think `onDateSelect` should be a better name for the prop, as the component is called `Datepicker`...

Reply:
Changed across the board to onDateChange